tests/integration/sync/test_sync_infra.py

```diff
@@ -403,3 +404,42 @@ def _run_sync_and_validate_lambda_call(self, dependency_layer: bool, template_pa
             self.assertIn("sum", lambda_response)
             self.assertEqual(lambda_response.get("message"), "hello world")
             self.assertEqual(lambda_response.get("sum"), 12)
+
+
+@skipIf(SKIP_SYNC_TESTS, "Skip sync tests in CI/CD only")
+@parameterized_class([{"dependency_layer": True}, {"dependency_layer": False}])
+class TestSyncInfraWithEsbuild(SyncIntegBase):
+    @parameterized.expand(["code/before/template-esbuild.yaml"])
+    def test_sync_infra_esbuild(self, template_file):
+        set_experimental(ExperimentalFlag.Esbuild)
+
+        template_path = str(self.test_data_path.joinpath(template_file))
+        stack_name = self._method_to_stack_name(self.id())
+        self.stacks.append({"name": stack_name})
+
+        self._run_sync_and_validate_lambda_call(self.dependency_layer, template_path, stack_name)
```

Comment:
why are we running the same twice? are we checking for behavior?

Reply:
Remnant from reusing an existing test, will remove.